packages/shared/src/util/features.ts

```diff
@@ -114,3 +116,54 @@ export function validateFeatureValue(
 
   return value;
 }
+
+type StaleFeatureReason =
+  | "error"
+  | "draft-state"
+  | "no-active-envs"
+  | "no-rules"
+  | "no-active-exps"
+  | "all-exps-onesided";
+
+export function isFeatureStale(
+  feature: FeatureInterface,
+  linkedExperiments: ExperimentInterface[] | undefined = []
+): { stale: boolean; reason?: StaleFeatureReason } {
+  if (feature.linkedExperiments?.length && !linkedExperiments.length) {
+    // eslint-disable-next-line no-console
+    console.error("isFeatureStale: linkedExperiments not provided");
+    return { stale: false, reason: "error" };
+  }
+
+  const twoWeeksAgo = subWeeks(new Date(), 2);
+  // TODO why is dateUpdated a string?
+  const stale = new Date(feature.dateUpdated) < twoWeeksAgo;
+
+  if (!stale) return { stale };
+
+  if (feature.draft) return { stale, reason: "draft-state" };
+
+  const envSettings = Object.values(feature.environmentSettings ?? {});
+
+  const noneEnabled = !envSettings.some((e) => e.enabled);
+  if (noneEnabled) return { stale, reason: "no-active-envs" };
+
+  const rules = envSettings.map((e) => e.rules).flat();
+
+  if (rules.length === 0) return { stale, reason: "no-rules" };
+
+  // TODO check if there are 'active' rules and return early if os
+
+  const noExpsActive = !linkedExperiments.some((e) => e.status === "running");
+  if (noExpsActive) return { stale, reason: "no-active-exps" };
```

Comment:
Right now, it's returning `stale: true` if there are no active experiments.  We should invert this to return `stale: false` if there are active experiments.  Will make it much easier to combine stale checks for other rule types.

Reply:
Also, the `e.status === "running"` check should be replaced by `includeExperimentInPayload` since there are cases when even a stopped experiment would still be considered "active"

Reply:
Addressed